src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesProjectTreeProvider.cs

```diff
@@ -65,7 +65,16 @@ internal class DependenciesProjectTreeProvider :
             TasksService = tasksService;
             _treeTelemetryService = treeTelemetryService;
 
+            // Hook this so we can unregister the snapshot change event when the project unloads
             unconfiguredProject.ProjectUnloading += OnUnconfiguredProjectUnloading;
+
+            Task OnUnconfiguredProjectUnloading(object sender, EventArgs args)
```

Comment:
Is there a benefit (or cost?) to not changing this to be a normal void returning event handler?

Reply:
This is required per CPS's public API. It consumes tasks from all handlers and awaits them in parallel.